addons/mail_bot/static/src/js/notification_alert.js

```diff
@@ -1,22 +1,57 @@
 odoo.define('mail.NotificationAlert', function (require) {
 "use strict";
 
+const { _t } = require('web.core');
 var Widget = require('web.Widget');
 var widgetRegistry = require('web.widget_registry');
 
 // -----------------------------------------------------------------------------
 // Display Notification alert on user preferences form view
 // -----------------------------------------------------------------------------
 var NotificationAlert = Widget.extend({
-   template: 'mail.NotificationAlert',
-   /**
-    * @override
-    */
-   init: function () {
-      this._super.apply(this, arguments);
-      var hasRequest = this.call('mailbot_service', 'isRequestingForNativeNotifications');
-      this.isNotificationBlocked = window.Notification && window.Notification.permission !== "granted" && !hasRequest;
-   },
+    template: 'mail.NotificationAlert',
+    /**
+     * @override
+     */
+    init() {
+        this._super(...arguments);
+        const hasRequest = this.call('mailbot_service', 'isRequestingForNativeNotifications');
+        this.isNotificationBlocked = window.Notification && window.Notification.permission !== 'granted' && !hasRequest;
+        if (this.isNotificationBlocked) {
+            if (navigator.userAgent.includes('Chrome')) {
+                this.settingUrl = "chrome://settings/content/siteDetails?site=https%3A%2F%2F" + window.location.host;
+            } else if (navigator.userAgent.includes('Firefox')) {
+                this.settingUrl = "about:preferences#privacy";
+            } else if (navigator.userAgent.includes('Safari')) {
+                this.settingPath = _t("(path:Safari > Preferences > Website > Notifications)");
+            }
+        }
+    },
+    /**
+     * Copies the browser settings URL link to the clipboard.
+     *
+     * @override
+     */
+    async start() {
+        await this._super(...arguments);
+        if (this.isNotificationBlocked && this.settingUrl) {
+            const $clipboardBtn = this.$('.o_notificationAlert_urlClipboardButton').tooltip({
+                title: _t("Copied!"),
+                trigger: 'manual',
+                placement: 'right'
+            });
+            const clipboard = new ClipboardJS($clipboardBtn[0], {
+                text: () => this.settingUrl,
+                container: this.el,
+            });
+            clipboard.on('success', function(e) {
+                setTimeout(() => {
+                    $clipboardBtn.tooltip('show');
+                    setTimeout(() => $clipboardBtn.tooltip('hide'), 800);
+                });
```

Comment:
Why do you need to wrap all of this in a `setTimeout()`?

Reply:
If we remove setTimeout() then "copied" doesn't shown by clicking on button as it takes hide immediately after show so this is for tooltip('show'). If i am not wrong.

Reply:
Indeed it doesn't display the tooltip without this `setTimeout`, but I fail to understand what causes this issue. I have a feeling this is caused by some buggy stuffs with bootstrap tooltips and putting this `setTimeout` is a hack to make it work.

I would put a comment that briefly explain this `setTimeout()` is necessary otherwise `.tooltip('show')` does not work for some reasons.